subprojects/problems/src/main/java/org/gradle/api/problems/ProblemCategory.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.problems;
+
+import org.gradle.api.Incubating;
+import org.gradle.util.Path;
+
+import static java.lang.String.join;
+
+/**
+ * Represents a category of problems.
+ *
+ * @since 8.5
+ */
+@Incubating
+public class ProblemCategory {
```

Comment:
❌ For me this exposes too many implementation details. `org.gradle.api.problems.ProblemCategory` should be just an interface declaring the category contract. 
```
public interface ProblemCategory {
  boolean hasPluginId();
  getPluginId();
  String getNamespace();
  String getCategory();
  List<Sring> getSubCateogies();
```
With proper javadoc.

All other details should be moved to `DefaultProblemCategory` or, if we need an intermediate class `InternalProblemCategory`.